Controller/Heroes/Baccarat/CharacterCards/AceOfSwordsSwordsBaccaratCharacterCardController.cs

```diff
@@ -0,0 +1,158 @@
+                            base.GameController.ExhaustCoroutine(coroutine);
+                        }
+                        yield break;
+                    }
+                case 1:
+                    {
+                        //One target deals itself 1 energy damage.
+                        SelectCardDecision selectDecision = new SelectCardDecision(GameController, base.HeroTurnTakerController, SelectionType.DealDamage, base.FindCardsWhere(new LinqCardCriteria((Card c) => c.IsTarget)), cardSource: base.GetCardSource());
+                        IEnumerator coroutine2 = base.GameController.SelectCardAndDoAction(selectDecision, (SelectCardDecision decision) => base.DealDamage(decision.SelectedCard, decision.SelectedCard, 1, DamageType.Energy, cardSource: base.GetCardSource()));
+                        if (base.UseUnityCoroutines)
+                        {
+                            yield return base.GameController.StartCoroutine(coroutine2);
+                        }
+                        else
+                        {
+                            base.GameController.ExhaustCoroutine(coroutine2);
+                        }
+                        break;
+                    }
+                case 2:
+                    {
+                        //Each hero character may deal themselves 1 toxic damage to draw a card now.
+                        this.actedHeroes = new List<Card>();
+                        IEnumerable<Function> functionsBasedOnCard(Card c) => new Function[]
+                        {
+                            new Function(base.FindCardController(c).DecisionMaker, "Deal self 1 toxic damage to draw a card now.", SelectionType.DrawCard, () => this.DealDamageAndDrawResponse(c))
+                        };
+                        IEnumerator coroutine3 = base.GameController.SelectCardsAndPerformFunction(this.DecisionMaker, new LinqCardCriteria((Card c) => c.IsHeroCharacterCard && c.IsInPlayAndHasGameText && !c.IsIncapacitatedOrOutOfGame && !this.actedHeroes.Contains(c), "active hero character cards", false, false, null, null, false), functionsBasedOnCard, true, base.GetCardSource(null));
+                        if (base.UseUnityCoroutines)
+                        {
+                            yield return base.GameController.StartCoroutine(coroutine3);
+                        }
+                        else
+                        {
+                            base.GameController.ExhaustCoroutine(coroutine3);
+                        }
+                        break;
+                    }
+            }
+            yield break;
+        }
+        private IEnumerator DealDamageAndDrawResponse(Card card)
+        {
+            if (card != null)
+            {
+                IEnumerator coroutine = base.DealDamage(card, card, 1, DamageType.Toxic, false, false, false, null, null, null, false, null);
+                IEnumerator coroutine2 = base.DrawCard(card.Owner.ToHero());
```

Comment:
Please remove Default arguments taht you aren't specifically setting.

Reply:
Removed.